File: js/day02/day02.js
const input = require('./input.json');
const data = input.data;

const exerciseOne = () => {
    let directions = data;
    let horizontal = 0;
    let depth = 0;
    directions.forEach((item) =>  {
        let command = item.split(' ')[0];
        let value = Number(item.split(' ')[1]);
        if (command === 'forward') {
            horizontal = horizontal + value;
        } else if (command === 'down') {
            depth = depth + value;
        } else {
            depth = depth - value;
        }
    });
    console.log('current position: ' + horizontal + ' - ' + depth);
    console.log('multiplied: ' + horizontal * depth);
};

const exerciseTwo = () => {
    let directions = data;
    let horizontal = 0;
    let aim = 0;
    let depth = 0;
    directions.forEach(function(item) {
        let command = item.split(' ')[0];
        let value = Number(item.split(' ')[1]);
        switch (command) {
            case 'forward':
                horizontal = horizontal + value;
                depth = depth + ((aim || 0) * value);
                break;
            case 'down':
                aim = aim + value;
                break;
            case 'up':
                aim = aim - value;
                break;
        }
    });
    console.log('current position: ' + horizontal + ' - ' + depth);
    console.log('multiplied: ' + horizontal * depth);
};

exerciseOne();
exerciseTwo();

File: python/day02/day02.py
f = open('./input.txt', 'r')
data = []
for x in f:
    data.append(x.strip())

# Exercise 1
horizontal = depth = 0
for i in range(len(data)):
    command = data[i].split(' ')[0]
    value = int(data[i].split(' ')[1])
    if (command == 'forward'):
        horizontal += value
    if (command == 'down'):
        depth += value
    if (command == 'up'):
        depth -= value
print(horizontal * depth)

# Exercise 2
horizontal = aim = depth = 0
for i in range(len(data)):
    command = data[i].split(' ')[0]
    value = int(data[i].split(' ')[1])
    if (command == 'forward'):
        horizontal += value
        depth += aim * value
    if (command == 'down'):
        aim += value
    if (command == 'up'):
        aim -= value
print(horizontal * depth)